packages/backend/src/services/ProjectService.ts

```diff
@@ -1,39 +1,61 @@
 import {
+    ApiQueryResults,
     CreateWarehouseCredentials,
-    DbtProjectConfig,
     Explore,
     ExploreError,
+    isExploreError,
+    MetricQuery,
     Project,
     SessionUser,
-    WarehouseCredentials,
 } from 'common';
-import { LightdashConfig } from '../config/parseConfig';
 import { projectAdapterFromConfig } from '../projectAdapters/projectAdapter';
 import { ProjectAdapter } from '../types';
 import { ProjectModel } from '../models/ProjectModel';
 import { DbtLocalProjectAdapter } from '../projectAdapters/dbtLocalProjectAdapter';
 import { analytics } from '../analytics/client';
+import { errorHandler, NotExistsError } from '../errors';
+import { compileMetricQuery } from '../queryCompiler';
+import { buildQuery } from '../queryBuilder';
 
 type ProjectServiceDependencies = {
-    lightdashConfig: LightdashConfig;
     projectModel: ProjectModel;
 };
 
-// Will return project with a working adapter, credentials.
-// Interface does not care about profiles.yml vs. warehouse credentials etc.
 export class ProjectService {
-    projectConfig: DbtProjectConfig;
+    projectModel: ProjectModel;
 
-    projectAdapter: ProjectAdapter;
+    cachedExplores: Record<string, Promise<(Explore | ExploreError)[]>>;
 
-    warehouseCredentials: WarehouseCredentials | undefined;
+    projectLoading: Record<string, boolean>;
 
-    projectModel: ProjectModel;
+    projectAdapters: Record<string, ProjectAdapter>;
 
-    constructor({ lightdashConfig, projectModel }: ProjectServiceDependencies) {
-        [this.projectConfig] = lightdashConfig.projects;
+    constructor({ projectModel }: ProjectServiceDependencies) {
         this.projectModel = projectModel;
-        this.projectAdapter = projectAdapterFromConfig(this.projectConfig);
+        this.projectAdapters = {};
```

Comment:
Main change is that the project service holds a cache containing the:
 * project adapters
 * loading status of projects
 * and cached explores  (<- we might move these to the db in future)